unstructured/ingest/connector/mongodb.py

```diff
@@ -78,39 +90,209 @@ def to_dict(
+            / self.connector_config.collection
+            / f"{self.document_meta.document_id}.json"
+        )
+
+    def update_source_metadata(self, **kwargs):
+        if self.document is None:
+            self.source_metadata = SourceMetadata(
+                exists=False,
+            )
+            return
+        self.source_metadata = SourceMetadata(
+            date_created=self.document_meta.date_created,
+            exists=True,
+        )
+
+    @SourceConnectionError.wrap
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    @BaseSingleIngestDoc.skip_if_file_exists
+    def get_file(self):
+        pass
+
+    @property
+    def record_locator(self) -> t.Optional[t.Dict[str, t.Any]]:
+        return {
+            "host": self.connector_config.host,
+            "collection": self.connector_config.collection,
+            "document_id": self.document_meta.document_id,
+        }
+
+
+@dataclass
+class MongoDBIngestDocBatch(BaseIngestDocBatch):
+    connector_config: SimpleMongoDBConfig
+    ingest_docs: t.List[MongoDBIngestDoc] = field(default_factory=list)
+    list_of_ids: t.List[str] = field(default_factory=list)
+    registry_name: str = "mongodb_batch"
+
+    @property
+    def unique_id(self) -> str:
+        return ",".join(sorted(self.list_of_ids))
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def _get_docs(self):
+        """Fetches all documents in a collection."""
+        from bson.objectid import ObjectId
+
+        # I don't like this here. Would prefer to get client
+        # from MongoDBSourceConnector or SimpleMongoDBConfig
+        client = self.connector_config.generate_client()
+        collection = self.connector_config.get_collection(client)
+        # MondoDB expects a list of ObjectIds (yet won't take a list comprehension for some reason
+        list_of_object_ids = []
+        for x in self.list_of_ids:
+            list_of_object_ids.append(ObjectId(x))
+        return list(collection.find({"_id": {"$in": list_of_object_ids}}))
+
+    def get_files(self):
+        documents = self._get_docs()
+        for doc in documents:
+            ingest_doc = MongoDBIngestDoc(
+                processor_config=self.processor_config,
+                read_config=self.read_config,
+                connector_config=self.connector_config,
+                document_meta=MongoDBDocumentMeta(
+                    collection=self.connector_config.collection,
+                    document_id=str(doc.get("_id")),
+                    date_created=doc.get("_id").generation_time.isoformat(),
+                ),
+                document=doc,
+            )
+            ingest_doc.update_source_metadata()
+            del doc["_id"]
+            filename = ingest_doc.filename
+            flattened_dict = flatten_dict(dictionary=doc)
+            str_values = [str(value) for value in flattened_dict.values()]
+            concatenated_values = "\n".join(str_values)
```

Comment:
Not sure if a doc was ever started but would be good to remembed to bucket this in with elasticsearch as ewe explore alternatives on how to represent the json data without loosing some of the structure.